File: Pratice/factorial.c
#include<stdio.h>
int fact(int n){// factorial of the number using recursion
    if(n==0){
        return 1;
    }
   //  int fnm1=fact(n-1);
    int fn=n*fact(n-1);

   return  fn;
    
}
int main(){//factorial of the number using for loop 
    int n;
    scanf("%d",&n);
    // int fact=1;
    // for(int i=1; i<=n; i++ ){
    //     fact*=i;
    // }
    // printf("%d",fact);
    int value=fact(n);

    printf("Factorial of the %d is %d", n , value);

}

File: JavaContent/Factorial.java
public class Factorial {
    public static int fact(int n){
        if(n==0||n==1){
        return 1;
        }


        return n*fact(n-1);
    }
    public static int loop(int n) {
        
        int result=1;
        for(int i=1; i<=n; i++){
            result=result*i;

        }
        return result;
    }
    public static void main(String[] args) {
        int n=5;
        int ans=fact(n);
        int result=loop(n);
        System.out.println(ans);
        System.out.println(result);
    }
    
}
